### PR TITLE
Vocsegmentation broken package solution

### LINKED ISSUE #1616: VOC example is broken
## 🐛 Bug description

This PR https://github.com/pytorch/ignite/pull/1605 breakes Pascal VOC example due to bad import:

```python
import torch
from apex import amp
from dataflow.datasets import VOCSegmentationOpencv  # <---- imported before sys.path.insert
from py_config_runner.config_utils import TRAINVAL_CONFIG, assert_config, get_params
from py_config_runner.utils import set_seed
from utils import exp_tracking  # <---- imported before sys.path.insert
from utils.handlers import predictions_gt_images_handler  # <---- imported before sys.path.insert

...

# Adds "code" folder to python path
sys.path.insert(0, Path(__file__).parent.parent.as_posix())
```

## Environment

 - PyTorch Version (e.g., 1.4):
 - Ignite Version (e.g., 0.3.0):
 - OS (e.g., Linux):
 - How you installed Ignite (`conda`, `pip`, source):
 - Python version:
 - Any other relevant information:

### PR DESCRIPTION
**Pull request by sparkingdark**
Fixes #1616 against broken pull req #1605 which broken the import 

I used https://stackoverflow.com/a/16985066 to import the parent folders and it's run.

I have very little ram , can you check the clearml or mlflow tests are passing for this pr @vfdev-5 

I don't get any dataflow import error,only I don't have a setup for mlflow or clearml.

```
ences/segmentation/pascal_voc2012/code/scripts/training.py
Traceback (most recent call last):
  File "/home/debo/ignite/examples/references/segmentation/pascal_voc2012/code/scripts/training.py", line 17, in <module>
    from utils import exp_tracking  # <---- imported before sys.path.insert
  File "/home/debo/ignite/examples/references/segmentation/pascal_voc2012/code/utils/exp_tracking.py", line 145, in <module>
    "No experiment tracking system is setup. "
RuntimeError: No experiment tracking system is setup. Please, setup either MLflow, Polyaxon or ClearML. For more details see NOTES_*.md
```

Check list:

- [ ] New tests are added (if a new feature is added)
- [ ] New doc strings: description and/or example code are in RST format
- [ ] Documentation is updated (if required)

### CODE REVIEW

File: examples/references/segmentation/pascal_voc2012/code/scripts/training.py [hunk 1]
```diff
@@ -1,17 +1,29 @@
 # This a training script launched with py_config_runner
 # It should obligatory contain `run(config, **kwargs)` method
-
+from pathlib import  Path
+import os
 import sys
-from collections.abc import Mapping
-from pathlib import Path
 
+#importing the package parent
```
Comment (vfdev-5) [inline examples/references/segmentation/pascal_voc2012/code/scripts/training.py:7]: Please, remove useless comments

File: examples/references/segmentation/pascal_voc2012/code/scripts/training.py [hunk 2]
```diff
@@ -1,17 +1,29 @@
...
+#so no path conflict not happen
+
+PACKAGE_PARENT = '..'
```
Comment (vfdev-5) [inline examples/references/segmentation/pascal_voc2012/code/scripts/training.py:10]: This should be removed as the same is done below line 37

File: examples/references/segmentation/pascal_voc2012/code/scripts/training.py [hunk 3]
```diff
@@ -1,17 +1,29 @@
...
+SCRIPT_DIR = os.path.dirname(os.path.realpath(os.path.join(os.getcwd(), os.path.expanduser(__file__))))
+sys.path.append(os.path.normpath(os.path.join(SCRIPT_DIR, PACKAGE_PARENT)))
+
+from dataflow.datasets import VOCSegmentationOpencv 
 import torch
 from apex import amp
-from dataflow.datasets import VOCSegmentationOpencv
+#from dataflow.datasets import VOCSegmentationOpencv  # <---- imported before sys.path.insert
```
Comment (vfdev-5) [inline examples/references/segmentation/pascal_voc2012/code/scripts/training.py:17]: Uncomment and remove comment. Place this line below the line 37

File: examples/references/segmentation/pascal_voc2012/code/scripts/training.py [hunk 4]
```diff
@@ -1,17 +1,29 @@
...
 from py_config_runner.config_utils import TRAINVAL_CONFIG, assert_config, get_params
 from py_config_runner.utils import set_seed
-from utils import exp_tracking
-from utils.handlers import predictions_gt_images_handler
+from utils import exp_tracking  # <---- imported before sys.path.insert
```
Comment (vfdev-5) [inline examples/references/segmentation/pascal_voc2012/code/scripts/training.py:20]: Same here and the next line: should go below the line 37 and remove comments

File: examples/references/segmentation/pascal_voc2012/code/scripts/training.py [hunk 5]
```diff
@@ -1,17 +1,29 @@
...
+from utils.handlers import predictions_gt_images_handler  # <---- imported before sys.path.insert
+
+# Adds "code" folder to python path
```
Comment (vfdev-5) [inline examples/references/segmentation/pascal_voc2012/code/scripts/training.py:23]: This is unnecessary copy of the line 37

File: examples/references/segmentation/pascal_voc2012/code/scripts/training.py [hunk 6]
```diff
@@ -1,17 +1,29 @@
...
+sys.path.insert(0, Path(__file__).parent.parent.as_posix())
+
+print("executed")
```
Comment (vfdev-5) [inline examples/references/segmentation/pascal_voc2012/code/scripts/training.py:26]: Remove this

**Review by vfdev-5 — CHANGES_REQUESTED** [review]
Please, rework the PR and read carefully what I asked you to do in the issue. Otherwise, it wont work.

**Comment by sparkingdark** [comment]
Okay I will rework on the pr and do the appropriate changes.

File: examples/references/segmentation/pascal_voc2012/code/scripts/training.py (hunk 1 above)
Comment (sparkingdark) [reply examples/references/segmentation/pascal_voc2012/code/scripts/training.py:7]: removed.

File: examples/references/segmentation/pascal_voc2012/code/scripts/training.py (hunk 2 above)
Comment (sparkingdark) [reply examples/references/segmentation/pascal_voc2012/code/scripts/training.py:10]: removed it as well.

File: examples/references/segmentation/pascal_voc2012/code/scripts/training.py (hunk 3 above)
Comment (sparkingdark) [reply examples/references/segmentation/pascal_voc2012/code/scripts/training.py:17]: done

File: examples/references/segmentation/pascal_voc2012/code/scripts/training.py (hunk 6 above)
Comment (sparkingdark) [reply examples/references/segmentation/pascal_voc2012/code/scripts/training.py:26]: removed

File: examples/references/segmentation/pascal_voc2012/code/scripts/training.py (hunk 5 above)
Comment (sparkingdark) [reply examples/references/segmentation/pascal_voc2012/code/scripts/training.py:23]: removed

**Comment by sparkingdark** [comment]
Done with changes @vfdev-5

**Comment by vfdev-5** [comment]
It is better now but still code-formatting is failing : https://github.com/pytorch/ignite/pull/1636/checks?check_run_id=1893555709

Please, read again contributing guide to see what to use to fix locally the code-formatting.
You are also introduced unused `import os` and removed required import of Mapping.

**Comment by sparkingdark** [comment]
What is the required import for mapping ,any specific line of import ?

**Comment by sdesrozis** [comment]
I suppose that @vfdev-5 was talking about using directives to let `isort ` skip some specific imports
```
import module  # isort:skip
```

See https://pypi.org/project/isort/#description

**Comment by sparkingdark** [comment]
@sdesrozis am I need to import that module in training.py to stop reordering?

**Comment by sdesrozis** [comment]
I don't think so. It's an indicative comment understood by `isort` during reordering.

**Comment by sparkingdark** [comment]
Okay

**Comment by vfdev-5** [comment]
@sparkingdark do not modify other files than `training.py` from Pascal VOC example

**Comment by sparkingdark** [comment]
Can I make a clean pr to this.

**Comment by vfdev-5** [comment]
You can also `git push <fork> <branch> --force` here as well.

**Comment by sparkingdark** [comment]
closed this pr and made a clean pr #1642

**Pull request closed without merging**